File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        
        left  = 0 # buy
        right = 1 # sell
        profit = 0

        while right < len(prices):
            if prices[left] < prices[right]:
                currentProfit = prices[right] - prices[left] # = 4 - 1 = 3
                if currentProfit > profit:
                    profit = currentProfit
            else:
                left = right
            right += 1
        return profit
        

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    left  = 0 // buy
    right = 1 // sell
    profit = 0
    while (right < prices.length){
            if (prices[left] < prices[right]){
                currentProfit = prices[right] - prices[left] 
                if (currentProfit > profit){
                    profit = currentProfit
                }
            }else{
                left = right
            }
            right += 1
        }
        return profit
        
    
};
        

        